opensearch-api/lib/opensearch/api/actions/cat/all_pit_segments.rb

```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+module OpenSearch
+  module API
+    module Cat
+      module Actions
+        # Retrieves info of all PIT segments
+        #
+        # @option arguments [String] :format a short version of the Accept header, e.g. json, yaml
+        # @option arguments [List] :h Comma-separated list of column names to display
+        # @option arguments [Boolean] :help Return help information
+        # @option arguments [List] :s Comma-separated list of column names or column aliases to sort by
+        # @option arguments [Boolean] :v Verbose mode. Display column headers
+        # @option arguments [Hash] :headers Custom HTTP headers
+        def all_pit_segments(arguments = {})
+          headers = arguments.delete(:headers) || {}
+          arguments = arguments.clone
+
+
+          method = OpenSearch::API::HTTP_GET
+          path   = '_cat/pit_segments/_all'
```

Comment:
I might be missing it, but where is this API listed for cat?

Reply:
It can be found here: https://opensearch.org/docs/2.6/search-plugins/point-in-time-api/#pit-segments

[open_point_int_time](https://github.com/opensearch-project/opensearch-ruby/blob/main/opensearch-api/lib/opensearch/api/actions/open_point_in_time.rb) is a legacy opensearch endpoint.

Reply:
Ah sorry I missed the cat endpoints!

For the `open_point_in_time`, my question was if it is still supported by OpenSearch?

Reply:
Just did a mini-research on this: The old PIT API was part of [Elasticsearch `x-pack`](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/point-in-time-api.html), and not the core Elasticsearch (But it was added to the [core Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/8.6/point-in-time-api.html) after OpenSearch'd been forked). In other words, the old PIT API was not carried over to OpenSearch. You get a `no uri handler` error when hitting one of these endpoints on OpenSearch.

Reply:
Did a quick lookup through all clients, only php and ruby clients still support these APIs and rust has some references. I created issues in [php](https://github.com/opensearch-project/opensearch-php/issues/137) and [rust](https://github.com/opensearch-project/opensearch-rs/issues/135) for removing these. For ruby, lets remove in this PR.